### PR TITLE
add hdinsight api version 2024-08-01-preview

### PR DESCRIPTION
**Pull request by yaning-liu**
# ARM (Control Plane) API Specification Update Pull Request 

> [!TIP]
> Overwhelmed by all this guidance? See the `Getting help` section at the bottom of this PR description.

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

spec_pr_review_workflow_diagram

## Purpose of this PR

What's the purpose of this PR? Check the specific option that applies. This is **mandatory**!

  - [ ] New resource provider.
  - [x] New API version for an existing resource provider. (If API spec is not defined in TypeSpec, the PR should have been created in adherence to [OpenAPI specs PR creation guidance](https://aka.ms/azsdkdocs/createopenapispec)).
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix OpenAPI spec quality issues in S360.
  - [ ] Other, please clarify:
    - _edit this with your clarification_

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [x] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [x] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://aka.ms/azurerpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to the diagram Step 2, "ARM API changes review", for this PR.

## Additional information

<details>
<summary> Viewing API changes</summary>

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

</details>
<details>
<summary>Suppressing failures</summary>

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[suppressions guide](https://aka.ms/azsdk/pr-suppressions) to get approval.

</details>

## Getting help

- First, please carefully read through this PR description, from top to bottom. Please fill out the `Purpose of this PR` and `Due diligence checklist`.
- If you don't have permissions to remove or add labels to the PR, request `write access` per [aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories](https://aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories")
- To understand what you must do next to merge this PR, see the `Next Steps to Merge` comment. It will appear within few minutes of submitting this PR and will continue to be up-to-date with current PR state.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- For help with ARM review (PR workflow diagram Step 2), see https://aka.ms/azsdk/pr-arm-review.
- If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
  This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

### CODE REVIEW

**Comment by ramoka178** [comment]
1.Please have the first commit to be the same as existing apiversion. Subsequent commits can have the changes that are to be introduced. This helps to review the changes only. You can choose to abondon this PR and create a new one, or, change the commits.
2. Please fill in the details for https://github.com/Azure/azure-rest-api-specs/pull/29608#issue-2377598408
3. Please fix Swagger LintDiff errors and resubmit for review.

**Comment by yuanhe772** [comment]
As the comments from @ramoka178 have been addressed, our teammate helped remove `[ARMChangesRequested]` tag for the PR review to resume.

File: specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2024-08-01-preview/cluster.json [hunk 1]
```diff
@@ -0,0 +1,2297 @@
...
+          "enum": [
+            "Disabled",
+            "Enabled"
+          ],
+          "x-ms-enum": {
+            "name": "PrivateLink",
+            "modelAsString": true
+          }
+        },
+        "publicIpTag": {
```
Comment (madhura-np) [inline specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2024-08-01-preview/cluster.json:1279]: >publicIpTag [](http://example.com/codeflow?start=9&length=11)

Please update the examples to reflect the usage of this property.

File: specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2024-08-01-preview/extensions.json [hunk 2]
```diff
@@ -0,0 +1,828 @@
...
+          "final-state-via": "location"
+        }
+      },
+      "get": {
+        "tags": [
+          "Extensions"
+        ],
+        "description": "Gets the status of Azure Monitor Agent on the HDInsight cluster.",
+        "x-ms-examples": {
+          "Enable cluster monitoring": {
```
Comment (madhura-np) [inline specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2024-08-01-preview/extensions.json:381]: >Enable [](http://example.com/codeflow?start=11&length=6)

rename to "Get cluster monitoring status"

File: specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2024-08-01-preview/extensions.json [hunk 3]
```diff
@@ -0,0 +1,828 @@
...
+          }
+        }
+      },
+      "delete": {
+        "tags": [
+          "Extensions"
+        ],
+        "description": "Disables the Azure Monitor Agent on the HDInsight cluster.",
+        "x-ms-examples": {
+          "Enable cluster monitoring": {
```
Comment (madhura-np) [inline specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2024-08-01-preview/extensions.json:421]: >Enable [](http://example.com/codeflow?start=11&length=6)

rename to Disable cluster monitor agent

File: specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2024-08-01-preview/extensions.json [hunk 4]
```diff
@@ -0,0 +1,828 @@
...
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/AzureMonitorRequest"
```
Comment (madhura-np) [inline specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2024-08-01-preview/extensions.json:351]: >AzureMonitorRequest [](http://example.com/codeflow?start=37&length=19)

This API is same as AzureMonitor request., same params and same response. Is this intentional?

File: specification/hdinsight/resource-manager/readme.md [hunk 5]
```diff
@@ -110,6 +110,53 @@ directive:
...
+  - Microsoft.HDInsight/preview/2024-08-01-preview/locations.json
+  - Microsoft.HDInsight/preview/2024-08-01-preview/operations.json
+  - Microsoft.HDInsight/preview/2024-08-01-preview/privateEndpointConnections.json
+  - Microsoft.HDInsight/preview/2024-08-01-preview/privateLinkResources.json
+  - Microsoft.HDInsight/preview/2024-08-01-preview/scriptActions.json
+  - Microsoft.HDInsight/preview/2024-08-01-preview/virtualMachines.json
+
+suppressions:
+  - code: EvenSegmentedPathForPutOperation
+    reason: Keep compatibility with old API version.
```
Comment (madhura-np) [inline specification/hdinsight/resource-manager/readme.md:132]: >with old API version. [](http://example.com/codeflow?start=31&length=21)

Can you point which old api version the compatibility for which these suppressions are needed?

File: specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2024-08-01-preview/cluster.json (hunk 1 above)
Comment (yuanhe772) [reply specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2024-08-01-preview/cluster.json:1279]: Done in `\specification\hdinsight\resource-manager\Microsoft.HDInsight\preview\2023-08-15-preview\examples\CreateHDInsightClusterWithCustomNetworkProperties.json`

File: specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2024-08-01-preview/extensions.json (hunk 2 above)
Comment (yuanhe772) [reply specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2024-08-01-preview/extensions.json]: Done

File: specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2024-08-01-preview/extensions.json (hunk 3 above)
Comment (yuanhe772) [reply specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2024-08-01-preview/extensions.json]: Done

File: specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2024-08-01-preview/extensions.json (hunk 4 above)
Comment (yuanhe772) [reply specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2024-08-01-preview/extensions.json:351]: Yes it is. We are adding these APIs with the same parameters/responses to keep compatibility with AzureMonitor request.

File: specification/hdinsight/resource-manager/readme.md (hunk 5 above)
Comment (yuanhe772) [reply specification/hdinsight/resource-manager/readme.md:132]: As mentioned in previous response, there errors happening in AzureMonitorAgent APIs also occur in AzureMonitor APIs which we are trying to keep compatibility with.

**Comment by yuanhe772** [comment]
Suppressions are added for backward compatibility with the old API versions, therefore they're actually the correct behavior that doesn't violate Microsoft API guidelines, hence we request suppressions to be approved, ty

**Comment by madhura-np** [comment]
The suppression cannot be approved, because the anti-pattern is being repeated for a new type. Please attend ARM office hours for a discussion.

**Comment by yaning-liu** [comment]
Here is the [LintDiff report](https://github.com/Azure/azure-rest-api-specs/pull/29608/checks?check_run_id=26978183506) right before suppressions are added. Suppressions comes from both cases: 
- existed APIs from previous API versions (2023-08-15-preview), e.g. [azureMonitor](https://github.com/Azure/azure-rest-api-specs/blob/fac7cf9a04d5ce577b91273713e10305886fcd65/specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2024-08-01-preview/extensions.json#L179-L320).
- new APIs which have the same format with existed APIs. We added APIs of [azureMonitorAgent](https://github.com/Azure/azure-rest-api-specs/blob/fac7cf9a04d5ce577b91273713e10305886fcd65/specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2024-08-01-preview/extensions.json#L321-L462) which are designed to be the same with existed APIs [azureMonitor](https://github.com/Azure/azure-rest-api-specs/blob/fac7cf9a04d5ce577b91273713e10305886fcd65/specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2024-08-01-preview/extensions.json#L179-L320). You can see they are almost the same and have same LintDiff errors. 

Since compatibility is the first concern we had when adding those APIs, we added those suppressions rather than updating APIs.

CC @madhura-np @ramoka178

**Comment by yaning-liu** [comment]
Here is the examples of those new APIs which have LintDiff errors:
Examples:
1. EvenSegmentedPathForPutOperation
   PathForNestedResource
   PutGetPatchResponseSchema
   ProvisioningStateSpecifiedForLROPut
   PutRequestResponseSchemeArm
   PutResponseCodes
   ResponseSchemaSpecifiedForSuccessStatusCode
   DefaultErrorResponseSchema
   LroErrorContent: [EnableLinuxClusterAzureMonitorAgent](https://github.com/Azure/azure-rest-api-specs/blob/fac7cf9a04d5ce577b91273713e10305886fcd65/specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2024-08-01-preview/examples/EnableLinuxClusterAzureMonitorAgent.json)

2. XmsPageableForListCalls: [GetLinuxClusterAzureMonitorAgentStatus](https://github.com/Azure/azure-rest-api-specs/blob/fac7cf9a04d5ce577b91273713e10305886fcd65/specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2024-08-01-preview/examples/GetLinuxClusterAzureMonitorAgentStatus.json)

3. DeleteResponseCodes
   DefaultErrorResponseSchema
   LroErrorContent
   LroLocationHeader: [DisableLinuxClusterAzureMonitorAgent](https://github.com/Azure/azure-rest-api-specs/blob/fac7cf9a04d5ce577b91273713e10305886fcd65/specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2024-08-01-preview/examples/DisableLinuxClusterAzureMonitorAgent.json)

**Comment by yuanhe772** [comment]
> The suppression cannot be approved, because the anti-pattern is being repeated for a new type. Please attend ARM office hours for a discussion. https://aka.ms/armofficehours Changing to WaitForARMFeedback to notify author.

Hi @madhura-np, we've just got off the IDC ARM OH, and it was suggested that we should justify the suppressions with more detailed explanations and examples in the comment section, pls refer to @yaning-liu 's comments above ^.

**Comment by yaning-liu** [comment]
As discuss with @suhasaraos in the meeting:
- I added scope restrictions to those suppressions.
- Checked with my team that we had these endpoints (/clusters/{clusterName}/extensions/{extensionName} and
/clusters/{clusterName}/extensions/azureMonitorAgent) since the first API version (2015-03-01-preview) until today. It has been working fine.
Please help to approve the suppressions.

File: specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2024-08-01-preview/extensions.json [hunk 6]
```diff
@@ -0,0 +1,828 @@
...
+          "final-state-via": "location"
+        }
+      },
+      "get": {
+        "tags": [
+          "Extensions"
+        ],
+        "description": "Gets the status of Operations Management Suite (OMS) on the HDInsight cluster.",
+        "x-ms-examples": {
+          "Get cluster monitoring status": {
```
Comment (razvanbadea-msft) [inline specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2024-08-01-preview/extensions.json:97]: >Get cluster monitoring status [](http://example.com/codeflow?start=11&length=29)

You should avoid having duplicate example names

File: specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2024-08-01-preview/extensions.json [hunk 7]
```diff
@@ -0,0 +1,828 @@
...
+          }
+        }
+      },
+      "delete": {
+        "tags": [
+          "Extensions"
+        ],
+        "description": "Disables the Azure Monitor on the HDInsight cluster.",
+        "x-ms-examples": {
+          "Disable cluster monitoring": {
```
Comment (razvanbadea-msft) [inline specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2024-08-01-preview/extensions.json:279]: >Disable cluster monitoring [](http://example.com/codeflow?start=11&length=26)

You should avoid having duplicate example names

**Comment by yaning-liu** [comment]
comments have been resolved.

**Review by razvanbadea-msft — APPROVED** [review]

**Pull request merged**